File: F1Levier/app/src/main/java/com/djira/f1levier/adapter/EquipeAdapter.java
package com.djira.f1levier.adapter;

import android.app.Dialog;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.WindowManager;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.ListView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.fragment.app.FragmentActivity;
import androidx.recyclerview.widget.RecyclerView;

import com.djira.f1levier.CourseActivity;
import com.djira.f1levier.R;
import com.djira.f1levier.database.AppDB;
import com.djira.f1levier.entity.Equipe;
import com.djira.f1levier.entity.Participant;

import java.util.List;

public class EquipeAdapter extends RecyclerView.Adapter<EquipeAdapter.ViewHolder> {
    // Initialize variable
    private List<Equipe> dataList;
    private FragmentActivity context;
    private AppDB database;

    // Create constructor

    public EquipeAdapter(FragmentActivity context, List<Equipe> dataList) {
        this.context = context;
        this.dataList = dataList;
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        // Initialize view
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.list_row_equipe, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull final ViewHolder holder, int position) {
        // initialize main data
        Equipe data = dataList.get(position);
        // Initialize database
        database = AppDB.getInstance(context);
        //Set text on text view
        holder.btnEquipe.setText(data.getName());

        holder.btnEquipe.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Equipe d = dataList.get(holder.getAdapterPosition());
                Intent intent = new Intent(context, CourseActivity.class);
                intent.putExtra("EQUIPE_NAME", String.valueOf(d.getName()));
                context.startActivity(intent);
            }
        });


//        final ArrayAdapter<String> adapter = new ArrayAdapter<String>(context,
//                android.R.layout.simple_list_item_2, prenoms);
//        holder.mListView.setAdapter(adapter);

    }

    @Override
    public int getItemCount() {
        return dataList.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {

        Button btnEquipe;
        ListView mListView;
        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            btnEquipe = itemView.findViewById(R.id.text_view_equipe);
            //mListView = (ListView) itemView.findViewById(R.id.listView);

        }
    }
}

File: TP4/CourseCODEP25/app/src/main/java/com/utbm/nouassi/manou/coursecodep25/ui/course/EquipeAdapter.kt
package com.utbm.nouassi.manou.coursecodep25.ui.course

import android.graphics.Color
import android.graphics.Typeface
import android.text.SpannableString
import android.text.style.StyleSpan
import android.text.style.UnderlineSpan
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TableLayout
import android.widget.TextView
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import com.utbm.nouassi.manou.coursecodep25.R
import com.utbm.nouassi.manou.coursecodep25.database.entity.CourseAndEquipe
import com.utbm.nouassi.manou.coursecodep25.database.entity.EquipeAndCoureur
import java.util.*


class EquipeAdapter (private var equipeCourseList: ArrayList<EquipeAndCoureur>) :
    RecyclerView.Adapter<EquipeAdapter.ViewHolder>() {

    constructor(): this(ArrayList<EquipeAndCoureur>()){
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): EquipeAdapter.ViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.course_equipe_item, parent, false)
        return EquipeAdapter.ViewHolder(view)
    }

    override fun getItemCount(): Int {
        return equipeCourseList.size
    }

    override fun onBindViewHolder(holder: EquipeAdapter.ViewHolder, position: Int) {
        holder.bind( equipeCourseList.get(position) )
    }

    fun addAllEquipes(equipes: ArrayList<EquipeAndCoureur>){
        equipeCourseList.addAll(equipes)
        notifyDataSetChanged()
    }

    fun clear(){
        equipeCourseList.clear()
        notifyDataSetChanged()
    }

    open class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {

        private var _courseEquipeTableLayout = view.findViewById<TableLayout>(R.id.course_equipe_tableLayout)


        fun bind(get: EquipeAndCoureur) {
            var equipe = get.equipe
            val tableRow: View = LayoutInflater.from( _courseEquipeTableLayout.context ).inflate( R.layout.equipe_tablerow, null, false )


            var _titreTextView = tableRow.findViewById<TextView>(R.id.equipe_tablerow_titre)
            val spanString = SpannableString( "Equipe " + equipe.numero + ": " + equipe.niveau  )
            spanString.setSpan(StyleSpan(Typeface.BOLD), 0, spanString.length, 0)
                _titreTextView.setText(spanString)
            _courseEquipeTableLayout.addView(tableRow)
            get.coureurs.forEach{
                val tableRow: View = LayoutInflater.from( _courseEquipeTableLayout.context ).inflate( R.layout.equipe_tablerow, null, false )
                var _titreTextView = tableRow.findViewById<TextView>(R.id.equipe_tablerow_titre)

                _titreTextView.setText( it.nom + " - niveau = "+ it.niveau.toString() )

                _courseEquipeTableLayout.addView(tableRow)
            }

        }

    }
}